contract-message-predicate/src/main.sw

```diff
@@ -1,18 +1,94 @@
 predicate;
 
-use std::{assert::assert, hash::sha256, tx::tx_script_bytecode};
+use std::{
+    address::Address,
+    tx::{
+        INPUT_COIN,
+        INPUT_CONTRACT,
+        INPUT_MESSAGE,
+        OUTPUT_CHANGE,
+        OUTPUT_CONTRACT,
+        OUTPUT_VARIABLE,
+        b256_from_pointer_offset,
+        tx_gas_limit,
+        tx_input_pointer,
+        tx_input_type,
+        tx_inputs_count,
+        tx_output_type,
+        tx_outputs_count,
+    tx_script_bytecode}
+};
+
+use std::assert::assert;
+use std::hash::sha256;
+use std::contract_id::ContractId;
+
+/// Get the ID of a contract input
+fn input_contract_id(index: u8) -> ContractId {
+    // Check that input at this index is a contract input
+    assert(tx_input_type(index) == INPUT_CONTRACT);
+    let ptr = tx_input_pointer(index);
+    let contract_id_bytes = b256_from_pointer_offset(ptr, 128); // Contract ID starts at 17th word: 16 * 8 = 128
+
+    // TODO: Replace with actual contract id
+    ~ContractId::from(0xf5dbe963c235c1e54f8732f1ecdc955df2ad8db8c9ab58eea8e1338762bf8bc2) //~ContractId::from(contract_id_bytes)
+}
+
+/// Get the contract ID from a message input's data
+fn contract_id_from_message_input(index: u8) -> ContractId {
+    // TODO: Replace with actual message check once input messages are enabled in the sdk
+    assert(tx_input_type(index) == INPUT_COIN);
+    ~ContractId::from(0xf5dbe963c235c1e54f8732f1ecdc955df2ad8db8c9ab58eea8e1338762bf8bc2)/*
+    // Check that input at this index is a message input
+    assert(tx_input_type(index) == INPUT_MESSAGE);
```

Comment:
Does this need to be commented out? We're asserting that the index is an Coin, and then asserting that it's a Message, which is contradictory

Reply:
The formatter screwed up the formatting a bit. Notice the `/*` at the end of line 41. It's probably screwing up due to the missing semicolon